### PR TITLE
Add a Go implementation of the DNS->DoH forwarder

### PR DESCRIPTION
**Pull request by bemasc**
This is off by default, enabled via remote config
for an A/B test.

### CODE REVIEW

File: Android/app/src/main/java/app/intra/net/go/GoIntraListener.java [hunk 1]
```diff
@@ -88,6 +103,37 @@ public void onUDPSocketClosed(UDPSocketSummary summary) {
...
-    FirebaseAnalytics.getInstance(context).logEvent(Names.UDP.name(), event);
+    FirebaseAnalytics.getInstance(vpnService).logEvent(Names.UDP.name(), event);
+  }
+
+  private static final Map<Long, Status> goStatusMap = new HashMap<>();
+  static {
+    goStatusMap.put(Doh.Complete, Status.COMPLETE);
+    goStatusMap.put(Doh.SendFailed, Status.SEND_FAIL);
+    goStatusMap.put(Doh.HTTPError, Status.HTTP_ERROR);
+    goStatusMap.put(Doh.BadQuery, Status.INTERNAL_ERROR); // TODO: Add a Status
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/net/go/GoIntraListener.java]: nit: could you clarify what status you want to add?

Comment (bemasc) [reply Android/app/src/main/java/app/intra/net/go/GoIntraListener.java]: Done

File: Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java [hunk 2]
```diff
@@ -157,7 +157,8 @@ public synchronized void close() {
   }
 
   private static boolean useGoDoh() {
-    return FirebaseRemoteConfig.getInstance().getBoolean("use_go_doh");
+    return true;
+    //return FirebaseRemoteConfig.getInstance().getBoolean("use_go_doh");
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java]: Did you want to delete or revert this?

Comment (bemasc) [reply Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java]: Yep, thanks.  Done.

**Pull request merged**